### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by dgreene-r7**
Refactored a few places to make testing simpler as well.

### CODE REVIEW

File: test/core-api-v1.js [hunk 1]
```diff
@@ -147,4 +147,8 @@ describe('Core API v1', () => {
         done();
       });
   });
+
+  it('returns a 500 if any source plugins fail');
+
+  it('returns a 429 if any source plugins have a warning');
```
Comment (fmitchell-r7) [inline test/core-api-v1.js]: Is there a commit that fills in these tests?

Comment (dgreene-r7) [inline test/core-api-v1.js]: Not yet. The spec around how we handle source statuses is incomplete.

Right now the `ok` property in the object returned from `Source#status()` is a bool. Eventually we'll want something to determine whether it's in a `warn` state and once we figure out how to handle that, we can implement it in the sources and fill in these tests.

I *think* this only affects a consul health check for propsd and shouldn't be critical but I'm not sure. The answer is probably deep within the guts of the conqueso client.

Comment (fmitchell-r7) [inline test/core-api-v1.js]: :+1:

**Comment by fmitchell-r7** [comment]
Thanks for improving out test coverage numbers! :shipit:

File: test/core-api-v1.js (hunk 1 above)
Comment (dgreene-r7) [inline test/core-api-v1.js]: I filed the need to complete the spec to define how we'll write the code that these tests will test (whew, that was a mouthful) as issue #107.

**Pull request merged**